packages/developer-console-spa/src/components/lighthouse/CreateProjectForm.tsx

```diff
@@ -0,0 +1,89 @@
+import { Button, Form, FormGroup, FormHelperText, TextInput } from '@patternfly/react-core';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { createLightHouseProjects } from '../../services/lighthouse';
+import ExclamationCircleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-circle-icon';
+
+const CreateProjectForm = ( props: any ) => {
+    const [ projectName, setProjectName ] = useState('');
+    const [ repoURL, setRepoURL ] = useState('');
+    const [ baseBranch, setBaseBranch ] = useState( '' );
+    const [ isPrimaryLoading, setIsPrimaryLoading ] = useState( false );
+    const { register, handleSubmit, watch, formState: { errors } } = useForm( {
+        mode: 'onBlur'
+    } );
+
+    const createProject = () => {
+        setIsPrimaryLoading( true );
+        let project = {
+            name: projectName,
+            baseBranch: baseBranch,
+            externalUrl: repoURL
+        };
+        createLightHouseProjects( project ).then( res => {
+            setIsPrimaryLoading( false );
+            props.setSelectedProject(res);
+            props.setShowConfirmation(true);
+        });
+    }
+    return (
+        <Form onSubmit={ handleSubmit( createProject ) } className="lighthouse-projectform--margintop" >
+            <FormGroup
+                isRequired
+                fieldId=""
+                helperTextInvalid="Project Name is mandatory"
+                helperTextInvalidIcon={ <ExclamationCircleIcon /> }
+                validated={ errors.projectName ? "error" : "default"  }
+                label="Project Name">
+                <TextInput
+                    isRequired
+                    type="text"
+                    id="project-name"
+                    { ...register( "projectName", { required: true } ) }
+                    name="projectName"
+                    aria-describedby="projectName"
+                    value={ projectName }
+                    validated={ errors.projectName ? "error" : "default" }
+                    onChange={ ( value: string ) => setProjectName( value ) }
```

Comment:
You don't need to provide the `value` and `onChange` if you're using the react-hook-form. Use only one of these methods.

And use the `Controller` component from react-hook-form, as mentioned by Akhil in his comment. See the `AddAppForm` component for an example.